bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BMap.java

```diff
@@ -416,6 +428,40 @@ public String toString() {
         return nativeData;
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void attemptFreeze(CPU.FreezeStatus freezeStatus) {
+        writeLock.lock();
+        try {
+            if (this.type.getTag() == TypeTags.OBJECT_TYPE_TAG) {
+                throw new BLangFreezeException("freeze not allowed on '" + getType() + "'");
```

Comment:
With `getType()` we could include details about the object itself. This could be especially useful in case such as unconstrained maps where we do not know beforehand the possible element types that could cause freeze to fail.

For example, we could say
```cmd
'freeze()' not allowed on 'Person'
```
instead of
```cmd
'freeze()' not allowed on objects
```
WDYT?